File: src/types/gateway.ts
export interface GatewayStatus {
  running: boolean
  port: number
  pid?: number
  startedAt?: string
}

export interface ServiceStatus {
  label: string
  pid: number | null
  running: boolean
  description: string
  cli_installed: boolean
}

export interface GuardianStatus {
  auto_restart_count: number
  max_auto_restarts: number
  manual_hold: boolean
  last_seen_running: string | null
  running_since: string | null
  give_up: boolean
}

File: src-tauri/src/models/gateway.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct GatewayStatus {
    pub running: bool,
    pub port: u16,
    pub pid: Option<u32>,
    pub started_at: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct GuardianStatus {
    pub auto_restart_count: u32,
    pub max_auto_restarts: u32,
    pub manual_hold: bool,
    pub last_seen_running: Option<String>,
    pub running_since: Option<String>,
    pub give_up: bool,
}
